sa/_db/migrations/20170423163751_CreateAuthorizationsTable.sql

```diff
@@ -0,0 +1,31 @@
+
+-- +goose Up
+-- SQL in section 'Up' is executed when this migration is applied
+CREATE TABLE `authorizations` (
+  `id` bigint(20) NOT NULL AUTO_INCREMENT,
+  `token` mediumblob,
+  `identifierValue` varchar(255) NOT NULL,
+  `identifierType` int NOT NULL,
+  `registrationID` bigint(20) NOT NULL,
+  `status` int NOT NULL,
+  `expires` datetime NOT NULL,
+  `allowedChallenges` tinyint,
+  `validated` datetime NOT NULL,
+  `validationRecord` mediumblob,
+  `error` mediumblob,
+  `thumbprint` mediumblob,
+
+  PRIMARY KEY (`id`),
+  UNIQUE KEY `identifier_key` (`identifierType`, `identifierValue`),
+
+  KEY `registrationID_status_expires_idx` (`registrationID`, `status`, `expires`),
+  CONSTRAINT `registrationID_status_expires_idx` FOREIGN KEY (`registrationID`) REFERENCES `registrations` (`id`) ON DELETE NO ACTION ON UPDATE NO ACTION,
```

Comment:
You've duplicated the `CONSTRAINT` clauses, one for each `KEY`, but `CONSTRAINT` is a property of a table, not of a key. Specifically, we want one constraint for the `authorizations` table that says "When you create a row in `authorizations`, it must have a `registrationID` that matches the `id` field of some row in the `registrations` table. That is, you should de-duplicate the `CONSTRAINT`s here, and name the one remaining `CONSTRAINT` to reflect its nature, e.g. `CONSTRAINT registrationID`.